### DISCUSSION TITLE
How to use WaitFor in a step

### DISCUSSION THREAD

**Discussion by ChrisBertrand** [danielgerlag/workflow-core#1151] [Q&A]
How do I correctly implement WaitFor in a step?

I've got the following step that waits for a event before completing.

`  return ExecutionResult.WaitForEvent("Event", "0", DateTime.Now);`

and then a API call that uses:

`await _workflowController.PublishEvent("Event", "0", "", DateTime.Now);`

but this doesn't complete the step and move onto the next in the Workflow.

```
builder.UseDefaultErrorBehavior(WorkflowErrorHandling.Suspend)
            .StartWith<StartStep>()
            .Then<WaitForStep>()
            .Then<EndStep>();
```

This always sits at the WaitFor step. What am I doing wrong?

**Comment by cjundt** [comment]
Hi,
Make sure you don't publish an event before subscription for this event is created.
You could use : `host.PersistenceStore.GetSubscriptions` for example.
